CHANGELOG.md

```diff
@@ -34,6 +34,27 @@
     * any analysis option specified on the command line (e.g. `--strong` or `--no-strong`) 
       takes precedence over any corresponding value specified in the analysis options file.
 
+* Pub
+
+  * `pub publish`
+
+    * Warns if a package imports a package that's not a dependency from within
+      `lib/` or `bin/`, or a package that's not a dev dependency from within
+      `test/` or `tool/`.
```

Comment:
I honestly overlooked these (and I guess so did Natalie).

Would be fairly simple to fix in a follow-up (I guess actually you might want to just say lib/ and bin/ are dependencies and any other folder is a dev_dependency)